frontend/src/app/shared/components/searchable-project-list/searchable-project-list.service.ts

```diff
@@ -4,20 +4,30 @@ import {
   ApiV3ListParameters,
   listParamsString,
 } from 'core-app/core/apiv3/paths/apiv3-list-resource.interface';
-import { BehaviorSubject } from 'rxjs';
+import { BehaviorSubject, combineLatest, forkJoin, of, Observable } from 'rxjs';
 import { IProject } from 'core-app/core/state/projects/project.model';
-import { getPaginatedResults } from 'core-app/core/apiv3/helpers/get-paginated-results';
 import { IHALCollection } from 'core-app/core/apiv3/types/hal-collection.type';
-import { finalize, take } from 'rxjs/operators';
+import { debounceTime, defaultIfEmpty, map, shareReplay, switchMap, take } from 'rxjs/operators';
 import { ApiV3Service } from 'core-app/core/apiv3/api-v3.service';
+import { ApiV3Filter } from 'core-app/shared/helpers/api-v3/api-v3-filter-builder';
 import { HttpClient } from '@angular/common/http';
 import { ID } from '@datorama/akita';
 import { IProjectData } from './project-data';
 import { CurrentProjectService } from 'core-app/core/current-project/current-project.service';
 
+const UNDISCLOSED_ANCESTOR = 'urn:openproject-org:api:v3:undisclosed';
+
 @Injectable()
 export class SearchableProjectListService {
+  // preferred size is chosen, so that it should be more than enough results for a user, while still being substantially
+  // smaller (and faster) than the maximum we could fetch at once
+  private preferredPageSize = 200;
+  private maximumPageSize = 1000;
```

Comment:
TIL: Apparently the maximum page size is configurable by admins. This means we can't assume that 1000 items per page is possible.